Fix kwargs and scoping in `@polynomial_ring`

mgkurtz opened the pull request:
See the new tests for examples now working.

fingolfin commented:
Thanks! Could you please also add a test that verifies it works now (and that it *stays* working)? It shouldn't be too hard, you can just do what the issues describes.

mgkurtz commented:
Added several tests. Rewrote keyword argument handling to fix #1631 also in the case of `polynomial_ring(QQ, ["x", "y"], internal_ordering=:lex)` with comma instead of semicolon and to handle `polynomial_ring(K, ["x", "y"]; internal_ordering=ordering)` with `K` and `ordering` some variables to be evaluated in current scope.

fingolfin reviewed (approved):
Awesome, thank you for the quick fix!
